Dockerfile

```diff
@@ -0,0 +1,9 @@
+FROM node:16-alpine
+
+WORKDIR /app
+
+COPY . /app
+
+RUN npm i
+
+ENTRYPOINT ["./node_modules/.bin/pm2", "start", "--no-daemon", "ecosystem.config.js"]
```

Comment:
Is running multiple processes (both bot and server) behind a process manager going to become an issue in the long term?
I'm thinking about things like logs (how will we separate the logs for both services) and scaling (what if we need to scale these independently of each other).

Generally, I expect a docker container to only run a single process.

Another option would be to build a separate image for the bot and server and run them independently. I don't fully understand the scope of this bot so maybe the extra effort/ overhead isn't worth it here.

Reply:
Optimizations can be made here, but it may take some time to modify.
Leave it as is for now.